File: LP-2/JAVA/chatbot.java
import java.util.*;
import java.util.regex.*;

public class GroceryChatbot {

    // Define patterns and responses
    static Map<String, String> responses = new HashMap<>();

    static {
        responses.put("\\b(hey|hi|hello)\\b", "Hey there! How can I help you?");
    }

    // Chatbot method
    public static String chatbot(String userInput) {
        String input = userInput.toLowerCase();

        for (Map.Entry<String, String> entry : responses.entrySet()) {
            Pattern pattern = Pattern.compile(entry.getKey());
            Matcher matcher = pattern.matcher(input);
            if (matcher.find()) {
                return entry.getValue();
            }
        }
        return "I'm sorry, I don't understand you. Please try again.";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Welcome to GROCERY Chatbot");

        while (true) {
            System.out.print("You: ");
            String userMsg = sc.nextLine();

            if (userMsg.equalsIgnoreCase("bye")) {
                System.out.println("Chatbot: Goodbye");
                break;
            }

            String response = chatbot(userMsg);
            System.out.println("Chatbot: " + response);
        }

        sc.close();
    }
}

File: LP-2/PYTHON/chatbot.py
import re

responses = {
    r"\b(hey|hi|hello)\b" : "Hey there ! how can i help you?"
}

def chatbot(user):
    user = user.lower()
    for pattern, response in responses.items():
        if re.search(pattern, user) :
            return response
    return "I'm sorry, I don't understand you. Please try again."

print("Welcome to GROCERY Chatbot")

while True:
    user_msg = input("You:")
    if user_msg.lower() == ["bye"]:
        print("Chatbot: Goodbye")
        break
    response = chatbot(user_msg)
    print("Chatbot:", response)